optics-core/src/Optics/Internal/Optic/TypeLevel.hs

```diff
@@ -111,12 +111,12 @@ data IxEq i is js where
 class AppendIndices xs ys ks | xs ys -> ks where
   appendIndices :: IxEq i (Curry xs (Curry ys i)) (Curry ks i)
 
--- | If the second list is empty, we can shortcircuit and pick the first list
--- immediately.
+-- | If the second list is empty, we can pick the first list
+-- even if nothing is known about it.
 instance {-# INCOHERENT #-} AppendIndices xs '[] xs where
```

Comment:
How about making this one similar to the one below?

```suggestion
instance {-# INCOHERENT #-} xs ~ zs => AppendIndices xs '[] zs where
```
?

Reply:
That doesn't look like a bad idea either.